### PR TITLE
add session expiration

### PR DESCRIPTION
**Pull request by algys**

### CODE REVIEW

**Comment by algys** [comment]
Issue:
- https://daocasino.atlassian.net/browse/DPM-49

File: sdk/scripts/gen_abi.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+echo "WARN: need to run from project root"
+
+eosio-abigen examples/simple/src/simple.cpp --contract=game --output=sdk/files/game.abi \
+    --extra-arg="-Isdk/include" \
+    --extra-arg="-Iexamples/simple/include" \
+    --extra-arg="-Isdk/libs/platform-contracts/contracts/platform/include" \
+    --extra-arg="-Isdk/libs/platform-contracts/contracts/events/include" \
+    --extra-arg="-Isdk/libs/platform-contracts/contracts/casino/include" \
+    --extra-arg="-I/usr/opt/eosio.cdt/1.6.3/include/eosiolib/capi" \
+    --extra-arg="-I/usr/opt/eosio.cdt/1.6.3/include/eosiolib/core" \
```
Comment (justefg) [inline sdk/scripts/gen_abi.sh]: /usr/opt
won't work on my system..

File: sdk/include/game-contract-sdk/game_base.hpp [hunk 2]
```diff
@@ -320,20 +314,39 @@ class game: public eosio::contract {
...
     }
 
-    GAME_ACTION(init)
-    void init(name platform, name events) {
+    CONTRACT_ACTION(close)
+    void close(uint64_t req_id) {
+        auto session_itr = sessions.require_find(req_id, "session with this req_id not found");
+        eosio::check(is_expired(req_id), "session isn't expired, only expired session can be closed");
+
+        eosio::action(
```
Comment (justefg) [inline sdk/include/game-contract-sdk/game_base.hpp:328]: better make a transfer method

```
void transfer(name to, asset quantity) {
    eosio::action(
        eosio::permission_level{_self, "active"_n},
        "eosio.token"_n,
        "transfer"_n,
        std::make_tuple(_self, to, quantity, std::string("player winnings"))
    ).send();
}
```

Comment (algys) [reply sdk/include/game-contract-sdk/game_base.hpp:328]: fixed in https://github.com/DaoCasino/game-contract-sdk/pull/3

File: sdk/scripts/gen_abi.sh (hunk 1 above)
Comment (algys) [reply sdk/scripts/gen_abi.sh]: fixed https://github.com/DaoCasino/game-contract-sdk/pull/2/commits/2db7a997dcb8c15bb77b37b0500ed8e334a92dd6

**Review by justefg — APPROVED** [review]

**Pull request merged**